server/models/role.ts

```diff
@@ -22,10 +22,10 @@ module.exports = (sequelize, DataTypes) => {
     validate: {
       // roles should only have 1 of these properties
       eitherOffchainOrOnchain() {
-        if (!(this.chain_id === undefined || this.offchain_community_id === undefined)) {
+        if (!(this.chain_id === null || this.offchain_community_id === null)) {
           throw new Error('Either chain_id or offchain_community_id!');
         }
-        if (this.chain_id !== undefined && this.offchain_community_id !== undefined) {
+        if (this.chain_id !== null && this.offchain_community_id !== null) {
```

Comment:
Is it possible for undefined to be passed here? Or just null?

Reply:
The db stores the empty column as 'null' not 'undefined'. So in validation, undefined is never passed.